### PR TITLE
Introduce Spark Failure Task Debug feature preview.

### PR DESCRIPTION
**Pull request by wezhang**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/hdinsight/templates/scala/sample/AgeMean_Div0.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+object AgeMean_Div0 {
+
+  def main(args: Array[String]) {
+    val sparkconf = new SparkConf().setAppName("SQL Query with issue")
+
+    val sc = new SparkContextWithFailureSave(sparkconf)
+    val spark = SparkSession
+      .builder()
+      .appName("Spark Age Mean")
+      .getOrCreate()
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/resources/hdinsight/templates/scala/sample/AgeMean_Div0.scala]: If we use SparkContext, SparkSession is useless here?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java [hunk 2]
```diff
@@ -48,6 +48,8 @@ public String getName() {
                 return "Spark on HDInsight (Scala)";
             case ScalaClusterSample:
                 return "Spark on HDInsight Sample (Scala)";
+            case ScalaFailureTaskDebugSample:
+                return "Spark with Failure Task Debugging on HDInsight Sample (Scala)";
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java]: >on HDInsight  [](start = 58, length = 13)

Remove "on HDInsight"? Too wordy.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java [hunk 3]
```diff
@@ -62,6 +64,8 @@ public String getDescription() {
                 return "HDInsight Spark blank module.";
             case ScalaClusterSample:
                 return "HDInsight Spark samples written in Scala.";
+            case ScalaFailureTaskDebugSample:
+                return "HDInsight Spark samples with Failure Task Debugging enabled library included, written in Scala";
```
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java]: > library included [](start = 83, length = 17)

Remove "library included"?

File: Utils/spark-tools/src/main/scala/org/apache/spark/FailureTask.scala [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+case class BroadcastValue(
+  id: Long,
+  value: String
+)
+
+case class ShuffleData(
+  name: String,
+  path: String,
+  location: String
+)
```
Comment (djlan) [inline Utils/spark-tools/src/main/scala/org/apache/spark/FailureTask.scala]: NIT: space

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java (hunk 2 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java]: Fix it

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java (hunk 3 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/HDInsightProjectTemplate.java]: Fix it.

File: PluginsAndFeatures/azure-toolkit-for-intellij/resources/hdinsight/templates/scala/sample/AgeMean_Div0.scala (hunk 1 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/resources/hdinsight/templates/scala/sample/AgeMean_Div0.scala]: The created SparkContext is to force SparkSession to use the SparkContextWithFailureSave one. Since we use the DataFrames API later, the SparkSession needs to be initialized here.

```
// For implicit conversions like converting RDDs to DataFrames
import spark.implicits._
```

File: Utils/spark-tools/src/main/scala/org/apache/spark/FailureTask.scala (hunk 4 above)
Comment (wezhang) [reply Utils/spark-tools/src/main/scala/org/apache/spark/FailureTask.scala]: To fix it with refactoring.

**Comment by wezhang** [comment]
```
Creating artifacts directory artifacts
+mvn install -f /home/travis/build/Microsoft/azure-tools-for-java/Utils/pom.xml -Dmaven.repo.local=/home/travis/build/Microsoft/azure-tools-for-java/.repository -q
Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: commons-io:commons-io:jar -> duplicate declaration of version 2.5 @ com.microsoft.azuretools:utils:3.6.0, /home/travis/build/Microsoft/azure-tools-for-java/Utils/pom.xml, line 353, column 25
[FATAL] Non-resolvable parent POM for com.microsoft.azuretools:spark-tools:0.1.0: Could not find artifact com.microsoft.azuretools:utils:pom:3.5.0 in central (http://repo.maven.apache.org/maven2) and 'parent.relativePath' points at wrong local POM @ line 3, column 13
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: commons-io:commons-io:jar -> duplicate declaration of version 2.5 @ com.microsoft.azuretools:utils:3.6.0, /home/travis/build/Microsoft/azure-tools-for-java/Utils/pom.xml, line 353, column 25
[WARNING] 'dependencyManagement.dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: commons-io:commons-io:jar -> duplicate declaration of version 2.5 @ line 353, column 25
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project com.microsoft.azuretools:spark-tools:0.1.0 (/home/travis/build/Microsoft/azure-tools-for-java/Utils/spark-tools/pom.xml) has 1 error
[ERROR]     Non-resolvable parent POM for com.microsoft.azuretools:spark-tools:0.1.0: Could not find artifact com.microsoft.azuretools:utils:pom:3.5.0 in central (http://repo.maven.apache.org/maven2) and 'parent.relativePath' points at wrong local POM @ line 3, column 13 -> [Help 2]
```

**Comment by wezhang** [comment]
Only lefts the expect build error:
```
> +--- com.microsoft.azuretools:spark-tools:0.1.0
+ echo ALERT: New dependencies introduced!
ALERT: New dependencies introduced!
```

Need to merge by hand.
